File: openapi/code_samples/csharp/authentication/access_token/code.cs
using System;
using System.Threading.Tasks;
using System.Net.Http;
using System.Text;
using Newtonsoft.Json;

namespace WebAPIClient {
  class Program {

    static async Task Main(string[] args) {
      await ApiCall();
    }

    private static async Task ApiCall() {
      var json = JsonConvert.SerializeObject(new {
        client_id = "qwcLzXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX", 
        client_secret = "7kscVxQZ32-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX", 
        grant_type = "client_credentials", 
        audience = "INSERT_PRODUCT_SERVER_URL_HERE"
      });

      var message = new HttpRequestMessage(HttpMethod.Post, "https://auth.reloadly.com/oauth/token");
      
      message.Headers.TryAddWithoutValidation("Content-Type", "application/json");

      using
      var httpClient = new HttpClient();
      var response = await httpClient.SendAsync(message);
      var responseBody = await response.Content.ReadAsStringAsync();
      var result = JsonConvert.DeserializeObject < dynamic > (responseBody);

      Console.WriteLine(result);
    }

  }
}

File: openapi/code_samples/go/authentication/access_token/code.go
package main

import (
	"fmt"
	"io/ioutil"
	"net/http"
	"strings"
)

func main() {

	url := "https://auth.reloadly.com/oauth/token"
	method := "POST"

	payload := strings.NewReader(`{
	"client_id":"qwcLzXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
	"client_secret":"7kscVxQZ32-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
	"grant_type":"client_credentials",
	"audience":"INSERT_PRODUCT_SERVER_URL_HERE"
}`)

	client := &http.Client{}
	req, err := http.NewRequest(method, url, payload)

	if err != nil {
		fmt.Println(err)
		return
	}
	req.Header.Add("Content-Type", "application/json")

	res, err := client.Do(req)
	if err != nil {
		fmt.Println(err)
		return
	}
	defer res.Body.Close()

	body, err := ioutil.ReadAll(res.Body)
	if err != nil {
		fmt.Println(err)
		return
	}
	fmt.Println(string(body))
}
